### PR TITLE
Add `WorkloadClusterMasterMemoryUsageTooHigh` alert.

### PR DESCRIPTION
**Pull request by whites11**
Towards: https://github.com/giantswarm/roadmap/issues/1855

Page SEs when master nodes have high memory usage for at least 1h.

### Checklist

- [x] Update CHANGELOG.md
- [ ] Add [Unit tests](https://github.com/giantswarm/prometheus-rules/#testing)
- [ ] Follow [Alert structure](https://github.com/giantswarm/prometheus-rules/#how-alerts-are-structured)
- [ ] Consider [creating a dashboard](https://docs.giantswarm.io/getting-started/observability/grafana/custom-dashboards/) ([guidelines](https://intranet.giantswarm.io/docs/product/ux/guidelines/dashboards/)) (if it does not exist already) to help oncallers monitor the status of the issue.
- [x] Request review from oncall area, as well as team (e.g: `oncall-kaas-cloud` GitHub group).

### CODE REVIEW

File: helm/prometheus-rules/templates/alerting-rules/node.workload_cluster.rules.yml [hunk 1]
```diff
@@ -125,3 +125,14 @@ spec:
         team: {{ include "providerTeam" . }}
         topic: kubernetes
     {{- end }}
+    - alert: WorkloadClusterMasterMemoryUsageTooHigh
+      annotations:
+        description: '{{`Machine {{ $labels.instance }} memory usage is too high (less than 10% and 2G of allocatable memory).`}}'
+      expr: ((((node_memory_MemTotal_bytes{cluster_type="workload_cluster"} - node_memory_MemFree_bytes{cluster_type="workload_cluster"} - node_memory_Cached_bytes{cluster_type="workload_cluster"}) / (node_memory_MemTotal_bytes{cluster_type="workload_cluster"}) * 100)) > 90 and (node_memory_MemFree_bytes{cluster_type="workload_cluster"} + node_memory_Cached_bytes{cluster_type="workload_cluster"}) < 2147483648) and on (node) kube_node_role{role="master"}
```
Comment (hervenicol) [inline helm/prometheus-rules/templates/alerting-rules/node.workload_cluster.rules.yml:131]: Writing the expression on multiple lines helps a lot with reading it!

```suggestion
      expr: |-
        ( ( (
              ( node_memory_MemTotal_bytes{cluster_type="workload_cluster"}
              - node_memory_MemFree_bytes{cluster_type="workload_cluster"}
              - node_memory_Cached_bytes{cluster_type="workload_cluster"}
              )
              / (node_memory_MemTotal_bytes{cluster_type="workload_cluster"}
            ) * 100
          )
        ) > 90
        and
        ( node_memory_MemFree_bytes{cluster_type="workload_cluster"}
          + node_memory_Cached_bytes{cluster_type="workload_cluster"}
        ) < 2147483648) 
        and on (node) kube_node_role{role="master"}
```

**Review by hervenicol — APPROVED** [review]

File: helm/prometheus-rules/templates/alerting-rules/node.workload_cluster.rules.yml [hunk 2]
```diff
@@ -125,3 +125,14 @@ spec:
...
+      for: 60m
+      labels:
+        area: kaas
+        cancel_if_outside_working_hours: {{ include "workingHoursOnly" . }}
```
Comment (hervenicol) [inline helm/prometheus-rules/templates/alerting-rules/node.workload_cluster.rules.yml:135]: '{{ include "workingHoursOnly" . }}', makes this alert 24/7 except on openstack and CAPZ. Is it what you really want?

**Review by hervenicol — CHANGES_REQUESTED** [review]
This alert should have an associated opsrecipe.

**Comment by whites11** [comment]
> This alert should have an associated opsrecipe.

There is no opsrecipe in this case IMHO. The alert is for SEs to get a list of overloaded clusters and get in touch with customers to sort that out

File: helm/prometheus-rules/templates/alerting-rules/node.workload_cluster.rules.yml (hunk 2 above)
Comment (whites11) [reply helm/prometheus-rules/templates/alerting-rules/node.workload_cluster.rules.yml]: No, thank you

**Comment by whites11** [comment]
@silenteh @LolloneS you asked for this alert I think. Is there a process already to handle it? Do you need an opsrecipe?

**Review by hervenicol — APPROVED** [review]
Well, if the opsrecipe just gives a bit of context ("This alert is for SEs to get a list of overloaded clusters and get in touch with customers to sort that out") and a link to the existing process doc, that's great.
Don't need more than a few lines, but helps a lot understanding what's going on.

On top of that, now it's not waking up people at night I feel less concerned :smile:

**Review by QuantumEnigmaa — APPROVED** [review]
LGTM

**Comment by LolloneS** [comment]
LGTM, but:
- I agree there should be a minimal opsrecipe (is there one?)
- I'd like to see this communicated to SEs _before_ the alert starts firing

Finally, I guess in 90% of the scenarios "please double your masters in size"/"please move to memory intensive instance flavour" will be the outcome of this alert. In the end very few customers run significant workloads on master nodes.

**Comment by pipo02mix** [comment]
IMO having a good ops recipe is important. All of us have some tricks, checking etcd health or API load, so gathering all in the recipe will help the SE when it is paged. I am happy to collaborate with that

**Comment by paurosello** [comment]
I will prepare a small OpsRecipe that links to some dashboards to try to figure out why we have high memory usage but TLDR is (we need to resize masters yes)

**Review by paurosello — APPROVED** [review]
LGTM

**Comment by whites11** [comment]
waiting on an ops recipe

**Comment by pipo02mix** [comment]
I have created a PR and ping Pau/sebastian for additions -> https://github.com/giantswarm/handbook/pull/50

**Pull request merged**